langchain_tools_demo/main.py

```diff
@@ -95,6 +93,19 @@ async def chat_handler(request: Request, prompt: str = Body(embed=True)):
         raise HTTPException(status_code=500, detail=f"Error invoking agent: {err}")
 
 
+@app.post("/reset")
+async def reset(request: Request):
+    """Reset agent"""
+    global user_agents
+    tasks = []
+    for user_agent in user_agents.values():
+        tasks.append(asyncio.create_task(user_agent.client.close()))
+        del user_agent.agent
+    asyncio.gather(*tasks)
+    user_agents.clear()
+    request.session.clear()
```

Comment:
This logic looks like it will delete and close all the agents. I think what we want instead is to just wipe the information for that user/session. This probably means:

1. grabbing the uuid from the session
2. use the uuid to grab the user agents
3. close the agent
3. remove uuid/agent from `user_agents`
4. remove the uuid and messages from the session

Reply:
changed to only deleting the current user agent.